src/parser/mod.rs

```diff
@@ -7156,6 +7156,35 @@ impl<'a> Parser<'a> {
                 partition,
                 with_name,
             }
+        } else if self.dialect.supports_alter_table_update() && self.parse_keyword(Keyword::UPDATE)
+        {
+            let mut assignments = vec![];
+            loop {
+                let target = self.parse_assignment_target()?;
+                self.expect_token(&Token::Eq)?;
+                // NOTE: Maybe it's better to save the index before parse_subexpr to do a real look
```

Comment:
Ah so regarding the earlier comment, I think what we could do would be to make `self.parse_assignment` reusable:

```rust
pub fn parse_assignment(&mut self) -> Result<Assignment, ParserError> {
	self.parse_assignment_with_prec(self.dialect.prec_unknown())
}

fn parse_assignment_with_precedence(self, precedence: u8) -> Result<Assignment, ParserError> {
	let target = self.parse_assignment_target()?;
    self.expect_token(&Token::Eq)?;
	let value = self.parse_subexpr(precedence)?;
    Ok(Assignment { target, value })

}
```
That would make it possible to use `parse_comma_separated` as well, since the issue seems to be with the `IN PARTITION` precedence when parsing the assigned expr